cmd/lifecycle/analyzer.go

```diff
@@ -120,6 +120,30 @@ func (a *analyzeCmd) Privileges() error {
 		if err != nil {
 			return cmd.FailErr(err, "initialize docker client")
 		}
+	} else {
+		if a.platformAPIVersionGreaterThan06() {
+			err := a.verifyReadAccess(a.previousImage)
+			if err != nil {
+				return cmd.FailErr(err)
+			}
+			err = a.verifyReadAccess(a.runImageRef)
+			if err != nil {
+				return cmd.FailErr(err)
+			}
+			if len(a.additionalTags) > 0 {
+				for _, tag := range a.additionalTags {
+					err := a.verifyReadWriteAccess(tag)
+					if err != nil {
+						return cmd.FailErr(err)
+					}
+				}
+			}
+		} else {
```

Comment:
Please correct me if I'm wrong but this is a new validation for platform API <= 0.6, right?
Does it mean that a user that is already successfully using an old platform API version will start have failures after upgrading to the new lifecycle version?

Reply:
@jabrown85 correct me if I'm wrong but I don't think there's harm in doing this for earlier platform API. It would just mean that a build that was going to fail during export will now fail during analyze.

Reply:
Oh actually...  maybe I'm wrong - creating the app image would succeed, but then we'd fail to create the cache. So the build would be "failed" but you'd still have an image!

Reply:
Maybe just to be on the safe side, we should not validate registry access up front for earlier platform APIs. Curious to know what others think.

Reply:
@yaelharel , @natalieparellano right now the PR is running the validation with the following logic:
**API >= 0.7**
-    read access <run-image>
-    read access <previous-image>
-    read/write access to <tags>

**API < 0.7**
-    read/write access to <cache-image>

The <cache-image> tag was removed from earlier 0.7 version but there is a PR open to add it back, right? I think if we merge that one first and the <cache-image> flag is back in version 0.7, then we can just apply the new validation for this version and avoid breaking any previous functionality.

Reply:
I agree with @natalieparellano. I think that we should just add the validation for new platform API (>= 0.7).

Reply:
In the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/1ab080726bdfd9210f0b21fe75dae32a42697ef2) I added the validations only for API >= 0.7